x-pack/plugins/observability_ai_assistant/public/components/message_panel/message_text.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { v4 } from 'uuid';
+import React from 'react';
+import type { Node } from 'unist';
+import { css } from '@emotion/css';
+import type { Parent, Text } from 'mdast';
+import ReactMarkdown from 'react-markdown';
+import { EuiText } from '@elastic/eui';
+
+interface Props {
+  content: string;
+  loading: boolean;
+}
+
+const cursorCss = css`
+  @keyframes blink {
+    0% {
+      opacity: 1;
+    }
+    50% {
+      opacity: 0;
+    }
+    100% {
+      opacity: 1;
+    }
+  }
+
+  animation: blink 1s infinite;
+  width: 10px;
+  height: 16px;
+  vertical-align: middle;
+  display: inline-block;
+  background: rgba(0, 0, 0, 0.25);
+`;
+
+const cursor = <span className={cursorCss} />;
+
+const CURSOR = `{{${v4()}}`;
+
+const loadingCursorPlugin = () => {
```

Comment:
This is a little bit of a hack that allows us to inject a blinking cursor element while keeping it on the same line as the text.